File: Python/Programmars/Level_1/K번째 수.py
def solution(array, commands):
    answer = []
    array.insert(0, 0)
    for command in commands:
        i, j, k = command
        temp = array[i:j+1]
        temp.sort()
        answer.append(temp[k-1])
    return answer


print(solution([1, 5, 2, 6, 3, 7, 4], [
      [2, 5, 3], [4, 4, 1], [1, 7, 3]]))  # [5, 6, 3]


# ref: sorted() 함수
def solution(array, commands):
    answer = []
    for command in commands:
        i, j, k = command
        answer.append(list(sorted(array[i-1:j]))[k-1])
    return answer

File: JavaScript/Programmars/정렬/K번째 수.js
function solution(array, commands) {
  const answer = [];

  let partArr;
  for (let [i, j, k] of commands) {
    partArr = array.slice(i - 1, j).sort((a, b) => a - b);
    answer.push(partArr[k - 1]);
  }

  return answer;
}

console.log(
  solution(
    [1, 5, 2, 6, 3, 7, 4],
    [
      [2, 5, 3],
      [4, 4, 1],
      [1, 7, 3],
    ]
  )
); // [5, 6, 3]
